src/components/NotificationProvider/messageBuilder.tsx

```diff
@@ -0,0 +1,49 @@
+import {
+  NotificationAction,
+  NotificationType,
+  QueuedNotification,
+} from "./types";
+import React, { ReactNode } from "react";
+
+export const queue = (notification: NotificationType): QueuedNotification => {
+  return { state: { queuedNotification: notification } };
+};
+
+export const info = (
+  message: string | ReactNode,
```

Comment:
I think we can do without the `string | ReactNode`, here and in the other places, since a `ReactNode` can implicitly already be a string.
```suggestion
  message: ReactNode,
```